src/TimeService.Validation/LeaveTime/LeaveTimeIntervalValidator.cs

```diff
@@ -39,8 +39,8 @@ private bool ValidateInterval(DateTimeOffset? startTime, DateTimeOffset? endTime
 
       bool isEndTimeValid = endTime is null  //endTime is valid if it is not edited
         || (dbLeaveTime is null || dbLeaveTime.EndTime.Add(endTime.Value.Offset) >= thisMonthFirstDay  //checks that end time was in correct interval if leaveTime is edited
-          || (dbLeaveTime.EndTime.Add(endTime.Value.Offset) >= thisMonthFirstDay.AddMonths(-1) && userTimeNow.Day <= 5)
-        && (endMonthFirstDay == thisMonthFirstDay.AddMonths(-1) && userTimeNow.Day <= 5)  //checks that new end time is in correct interval
+          || (dbLeaveTime.EndTime.Add(endTime.Value.Offset) >= thisMonthFirstDay.AddMonths(-1) && userTimeNow.Day <= 5))
+        && ((endMonthFirstDay == thisMonthFirstDay.AddMonths(-1) && userTimeNow.Day <= 5)  //checks that new end time is in correct interval
```

Comment:
May be better to use < instead of <=?

Reply:
replaced 5 with int constant to make this clearer, and as its name says it should be 5
also there is no significant performance difference between < and <=